### DISCUSSION TITLE
api/auth/* not found on production

### DISCUSSION THREAD

**Discussion by curmande** [nextauthjs/next-auth#2469] [Help] [answered]
### Description 🐜

Hello,

it is 2 days now which i struggling with totally not working as expected issue.

I tried everything - configure my project correctly by documentation, start x new projects to test it but i am not successful at all.

Sometimes it doesn't work when i develop and sometimes it works - if it not works in develop, then i got error - client failed to fetch - err_too_many_redirects

But everytime when i export my build using "npm build && npm export" and try to make it work on server it ALWAYS end up on 

https://next-auth.js.org/errors#client_fetch_error session SyntaxError: Unexpected token < in JSON at position 0

which seems like bug in library or i really don't know what i am doing wrong 

Yes i have NEXTAUTH_URL in my .env + i using custom credentials but when i want to use signin in build it seems like the /api/auth/* is just not working and **** with 404

also thin happens on https or http - still the same

any advice ? 

Many thanks

### Is this a bug in your own project?

Yes

### How to reproduce ☕️

   <button onClick={() => signIn()}>Sign in</button>

### Screenshots / Logs 📽

![Untitled-1](https://user-images.githubusercontent.com/213240/127772196-6868a92d-d146-466f-aba6-e6dda608591c.png)
![Untitled-2](https://user-images.githubusercontent.com/213240/127772197-a94ef3ac-e51c-4dcf-a823-bcc9ba5d398a.png)
![Untitled-3](https://user-images.githubusercontent.com/213240/127772198-7d48adb5-f7ec-4bb7-ba0f-d2abf0edc96b.png)

### Environment 🖥

production

### Contributing 🙌🏽

Yes,  I am willing to help solve this bug in a PR

**Comment by balazsorban44** [comment, marked as answer]
Certainly a configuration issue. Do you have reproduction?

Make sure the .env file is present in production.

**Reply by curmande** [reply]
Okay,

i found out that the env variable was not seen in production but even when i repair it, it still not working 

here is my project 

https://github.com/curmande/testing-typescript-next-auth/tree/master

1) pull project
2) npm i
3) set env variable to http://localhost:3000 in files .env + next.config.js
4) npm run dev, then open http://localhost:3000 in browser - now you will see a sign button and env variable NEXTAUTH_URL - when i click on signin button, this used to redirect me on login page - now i am gettin redirect to error page and in the debug console flashes ERR_TOO_MANY_REDIRECTS right before redirect - i believe its something with browser but who knows
![Untitled-1](https://user-images.githubusercontent.com/213240/127856816-b8e7bac5-e6ca-42c9-8660-9f14233d389a.png)

Main problem 

5) set env variables NEXTAUTH_URL in files to the url of your server (http://25.22.141.186:8000 in my case)
6) npm run build - this will build a production build which will be in the build folder
7) copy content of build folder to the root of your server 
8) open server url 
9) now i could clearly see that NEXTAUTH_URL is there and set it right but in my debug console gettin the annoying
https://next-auth.js.org/errors#client_fetch_error session SyntaxError: Unexpected token < in JSON at position 0
![Untitled-2](https://user-images.githubusercontent.com/213240/127856835-ff5f357d-eec3-4d60-9eb5-e54556e2df33.png)


10) when i click on signin button i am at the error page 
![Untitled-3](https://user-images.githubusercontent.com/213240/127856836-999a4d40-0dff-4baf-b01c-1fed49005bdb.png)

expected behavior is that when i will click on signin button - i will be on the login page able to login 

If you make it work somehow, you definitely help me a lot 

Many thanks

**Reply by balazsorban44** [reply]
You don't need to set env variables in `next.config.js`. See https://nextjs.org/docs/basic-features/environment-variables

`NEXTAUTH_URL_INTERNAL` might be useful here as well maybe?: https://next-auth.js.org/configuration/options#nextauth_url_internal

**Reply by curmande** [reply]
Actually i don't see env variable from .env but if i set it in next.config.js -> then i see it also when i building or running there is clear message that the .env is loading

"info  - Loaded env from .../testing-typescript-next-auth-master/.env"

NEXTAUTH_URL_INTERNAL unfortunately doesn't help

Everything still the same

**Reply by curmande** [reply]
Also i need to mention that i was using "next build && next export" and my build is static site 

●  (SSG)     automatically generated as static HTML + JSON (uses getStaticProps)

Maybe it will help

**Reply by balazsorban44** [reply]
If I understand this correctly, next export disables API routes, meaning you won't be able to use `next-auth`

https://nextjs.org/docs/messages/api-routes-static-export

There is nothing for us to do in this situation

**Reply by curmande** [reply]
Okay, then i think we are done. I choose another method for authorizations in my app

Thank you for your time and help
